CMakeLists.txt

```diff
@@ -21,7 +21,11 @@ project("Bitcoin Core"
   LANGUAGES CXX C ASM
 )
 
-if(MSVC)
+# Configurable options.
+# When adding a new option, end the <help_text> with a full stop for consistency.
+option(CXX20 "Enable compilation in C++20 mode." OFF)
+
+if(CXX20 OR MSVC)
```

Comment:
nit: this would enable C++20 unconditionally on MSVC. Maybe the option should not exist on MSVC? For Windows users, it would be strange to have an option which does nothing and when CXX20=OFF, then we still compile in C++20 mode.

C++17 seems to be supported on MSCV in `configure.ac`.

Reply:
> C++17 seems to be supported on MSCV in `configure.ac`.

MSVC build configuration is defined in the `build_msvc` directory, not by `configure.ac`.

And C++20 for MSVC was introduced in 88ec5d40dcf5d9f95217b123b48203b2f334c0a1 from #25472.

Reply:
> Maybe the option should not exist on MSVC? 

Thanks! Going to fix this.

Reply:
> > Maybe the option should not exist on MSVC?
> 
> Thanks! Going to fix this.

[Updated](https://github.com/bitcoin/bitcoin/pull/27060#issuecomment-1424023133).